pkg/solana/chain_reader_test.go

```diff
@@ -35,3 +66,596 @@ func TestSolanaChainReaderService_ServiceCtx(t *testing.T) {
+		r.testStructQueue = append(r.testStructQueue, &nextStruct)
+
+		fallthrough
+	default:
+		if r.testStructQueue == nil || len(r.testStructQueue) == 0 {
+			r.test.FailNow()
+		}
+
+		nextTestStruct := r.testStructQueue[0]
+		r.testStructQueue = r.testStructQueue[1:len(r.testStructQueue)]
+
+		cdc := makeTestCodec(r.test, fmt.Sprintf(baseIDL, testStructIDL, strings.Join([]string{midLevelStructIDL, innerStructIDL}, ",")))
+		bts, err := cdc.Encode(ctx, nextTestStruct, "TestStruct")
+		if err != nil {
+			r.test.FailNow()
+		}
+
+		r.client.On("ReadAll", mock.Anything, mock.Anything).Return(bts, nil).Once()
+	}
+
+	return r.service.GetLatestValue(ctx, contractName, method, params, returnVal)
+}
+
+func (r *wrappedTestChainReader) Bind(ctx context.Context, bindings []types.BoundContract) error {
+	return r.service.Bind(ctx, bindings)
+}
+
+func (r *wrappedTestChainReader) CreateContractType(contractName, itemType string, forEncoding bool) (any, error) {
+	if AnyContractName+EventName == contractName+itemType {
+		// events are not supported, so just make the tests pass
+		return nil, types.ErrNotFound
+	}
+
+	return r.service.CreateContractType(contractName, itemType, forEncoding)
+}
+
+// SetLatestValue is expected to return the same bound contract and method in the same test
+// Any setup required for this should be done in Setup.
+// The contract should take a LatestParams as the params and return the nth TestStruct set
+func (r *chainReaderInterfaceTester) SetLatestValue(t *testing.T, testStruct *TestStruct) {
+	if r.reader == nil {
+		r.reader = &wrappedTestChainReader{
+			test:   t,
+			tester: r,
+		}
+	}
+
+	r.reader.testStructQueue = append(r.reader.testStructQueue, testStruct)
+}
+
+func (r *chainReaderInterfaceTester) TriggerEvent(t *testing.T, testStruct *TestStruct) {
+	t.Skip("Events are not yet supported in Solana")
+}
+
+func (r *chainReaderInterfaceTester) GetBindings(t *testing.T) []types.BoundContract {
+	return []types.BoundContract{
+		{Name: strings.Join([]string{AnyContractName, MethodTakingLatestParamsReturningTestStruct, "0"}, "."), Address: r.address[0], Pending: true},
+		{Name: strings.Join([]string{AnyContractName, MethodReturningUint64, "0"}, "."), Address: r.address[1], Pending: true},
+		{Name: strings.Join([]string{AnyContractName, DifferentMethodReturningUint64, "0"}, "."), Address: r.address[2], Pending: true},
+		{Name: strings.Join([]string{AnyContractName, MethodReturningUint64Slice, "0"}, "."), Address: r.address[3], Pending: true},
+		{Name: strings.Join([]string{AnyContractName, MethodReturningSeenStruct, "0"}, "."), Address: r.address[4], Pending: true},
+		{Name: strings.Join([]string{AnySecondContractName, MethodReturningUint64, "0"}, "."), Address: r.address[5], Pending: true},
+	}
+}
+
+func (r *chainReaderInterfaceTester) MaxWaitTimeForEvents() time.Duration {
+	// From trial and error, when running on CI, sometimes the boxes get slow
+	maxWaitTime := time.Second * 20
```

Comment:
not sure that's true for this, you're using a mock so it should return right away.  1s could be good until you have a real implementation.

Reply:
This will have no effect until events are implemented.